### ISSUE TITLE
Power button doesn't work correctly

### ISSUE THREAD

**Issue by samcday** [samcday/phrog#8]
Right now it seems to blank the screen, but there's no way to get anything back. Nothing seems to happen in logs. Mysterious.

Tested on both Silverblue laptop and pmOS tablet.

**Comment by samcday** [comment]
The power menu isn't showing up in phrog builds at all AFAICT. That seems quite related.

**Comment by samcday** [comment]
The power menu not showing up at all is [quite intentional](https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1055), so I guess I'll need to open an issue upstream and figure out how we might accommodate that. I assume Guido would concur that users might want, and should be able, to shutdown/reboot their device before they've even logged into a session.

I don't think the power menu's absence has anything to do with the main issue at hand here, though. The issue here seems to be that once the power button is pressed, the shell enters the `PHOSH_STATE_BLANKED` state, but subsequent interactions/button presses (including pressing the power button again) do *not* remove that state.

**Comment by agx** [comment]
Is there a gsd-power running?

**Comment by samcday** [comment]
Hm. There is not. I see that the gsd daemons are spawned as part of the ~~many-tentacled lovecraftian horror~~ sprawling user-level systemd units pulled in via the `foo-session.target`.

I tried to just fire up that one gsd-power daemon (pointing it at the `XDG_RUNTIME_DIR` + `DBUS_SESSION_BUS_ADDRESS` + I scooped out of `/proc/<greetd pid>/environ` but that didn't seem to help things. I probably need to do more than that, but this is a great thread to pull on further, thanks!

Assuming the gsd-power running is the missing piece here, it begs the question how this should get booted in a greetd session :thinking: 

(I also welcome any further tips/wisdom-nuggets that save me several hours xD)

**Comment by samcday** [comment]
I see that all the `gsd-` daemons are spawned by gdm, so I guess it really is going to be phrog's responsibility to fire these things up to get stuff working properly, eek!

**Comment by samcday** [comment]
Hm, I guess maybe this isn't so bad if I just setup phrog as a proper session and run it up via `gnome-session`...

**Comment by agx** [comment]
> Seems that's what GDM is doing as well anyways.

Yes, tht would be my suggestion as this would also simplify things like emergency calling, etc later on. Initially just `org.gnome.SettingsDaemon.MediaKeys`, `org.gnome.SettingsDaemon.Power`, `mobi.phosh.Shell;sm.puri.OSK` , `mobi.phosh.Shell` should be a good start. That would also make the media keys for e.g. brightness, volume, etc work and give proper OSDs for those.

**Comment by samcday** [comment]
I was tempted to close this out now since it'll be resolved by #70 ... but there's still the outstanding issue here that the shutdown action is not available in phrog right now, which will be resolved by https://gitlab.gnome.org/World/Phosh/phosh/-/issues/1164
